controllers/performanceprofile_controller.go

```diff
@@ -135,6 +142,91 @@ func (r *PerformanceProfileReconciler) SetupWithManager(mgr ctrl.Manager) error
 		Complete(r)
 }
 
+// uninstall PAO OLM operator and all of its artifacts
+// this should apply only from version 4.11
+func (r *PerformanceProfileReconciler) removeOLMOperator() error {
+	paoCSV := "performance-addon-operator.v4.10.0"
+	subscription := &olmv1alpha1.Subscription{}
+	key := types.NamespacedName{
+		Name:      "performance-addon-operator",
+		Namespace: "openshift-performance-addon-operator",
+	}
+
+	if err := r.Get(context.TODO(), key, subscription); err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		}
+	} else {
+		klog.Infof("Removing performance-addon-operator subscription %s", subscription.Name)
+		if subscription.Status.CurrentCSV != paoCSV {
+			klog.Warningf("Subscription to be removed contains a current CSV version %s which is different from %s", subscription.Status.CurrentCSV, paoCSV)
```

Comment:
probably we should exit here

Reply:
I wonder what can bring us to this scenario at all. maybe this is an overkill

Reply:
only if a user will not update a subscription from 4.9 or earlier version(but it should be the unsupported scenario, so I think we should exit with the error that first, you should upgrade to 4.10) and will upgrade to 4.11.